tests/5.1/target/test_target_map_with_present_modifier.c

```diff
@@ -0,0 +1,67 @@
+//===--- test_target_map_with_present_modifier.c ----------------------------------------------------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+//
+// This test checks tests the present map-type-modifier on a map clause. The test maps several
+// different data types to device with tofrom map-type and then checks for expected updated values on
+// the host.
+//
+////===--------------------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int errors; 
+
+int test_present_modifier() {
+
+  int i;
+  int scalar = 1;
+  int a[N];
+
+  struct {
+    int var;
+    int b[N];
+  } member; 
+
+  member.var = 1;
+  
+  for (i = 0; i < N; i++) { 
+    a[i] = i;
+    member.b[i] = i;
+  }
+
+#pragma omp target map (present, tofrom: scalar, a, member) 
```

Comment:
This testcase has a bare **target** with **present** modifier. Thus, the following applies:

> If a **map** clause appears on a **target**, **target data**, **target enter data** or **target exit data** construct with a **present** _map-type-modifier_ then on entry to the region if the corresponding list item does not appear in the device data environment then an error occurs and the program terminates.

As in sollve_vv, there is no real infrastructure to test for should-fail tests, I assume the testcase should run. But in that case, the **target** construct must either be inside a **target data** region that maps the variables or the data must be mapped before with **target enter data**. 

(Or a combination of both: some variables one way, some the other way.)

* For a **target exit data**, the **present** modifier could also be used as that should work, given the variables have been mapped before.
* For the **target** construct, an **always** modifier would be required in addition, if the map-to value has been modified between the 'target (enter) data' and the 'target' — and when the result should be checked before the 'target exit data' or insie the 'target data' region, respectively.

Reply:
Might be useful to refer to 5.1/target_update/test_target_update_to_present.c